<launch>
    <arg name="namespace"   default="$(env UAV_NAMESPACE)"/>
    <arg name="odom"        default="mavros/global_position/local"/>

    <arg name="x_offset"    default="visual_servo/offset_x"/>
    <arg name="y_offset"    default="visual_servo/offset_y"/>
    <arg name="z_offset"    default="visual_servo/offset_z"/>
    <arg name="x_target_error" default="visual_servo/target_error_x"/>
    <arg name="y_target_error" default="visual_servo/target_error_y"/>
    <arg name="z_target_error" default="visual_servo/target_error_z"/>
    <arg name="process_value" default="visual_servo/process_value"/>
    <arg name="status"      default="visual_servo/is_enabled"/>

    <group ns="$(arg namespace)">
        
        <!-- Visual servo state machine parameters -->
        <rosparam command="load"
            file="$(find uav_ros_control)/config/reference/visual_servo_pursuit_sm.yaml" />

        <!-- Visual Servo state machine node -->
        <node
            pkg="uav_ros_control"
            type="visual_servo_sm_pursuit"
            name="visual_servo_state_machine"
            output="screen">

            <remap from="odometry" to="$(arg odom)"/>
        </node>

                 <!-- Visual servo state machine parameters -->
        <rosparam command="load"
            file="$(find uav_ros_control)/config/reference/visual_servo_pursuit.yaml" />
        
        <!-- Visual servo node -->
    </group>
</launch>
